### PR TITLE
Resolve Misleading Timestamp Issue When Quoting Old Messages

### LINKED ISSUE #5308: Misleading timestamp when quoting old messages
### Description:

When quoting old messages, the timestamp of the quote shows the time instead of the date. This is misleading and in contrast to what happens in the browser version.

### Environment Information:

- Rocket.Chat Server Version: 6.4.4
- Rocket.Chat App Version: 4.42.0.50842
- Device Name: ELE-L29 (Huawei P30)
- OS Version: EMUI 12.0.0 (Android 11)

### Steps to reproduce:

1. Write a message quoting an old message.

### Expected behavior:

The timestamp in the quote shows a date, just as in the browser:

image

### Actual behavior:

The timestamp in the quote shows only the time of the day, leading to the impression that the quoted message was written less than 24h hours ago:

image

### PR DESCRIPTION
**Pull request by Ansh101112**
## Proposed changes
This pull request addresses issue #5308. The issue reported a misleading timestamp when quoting old messages. The timestamp in the quoted message was displaying only the time, which was inconsistent with the behavior in the browser version. To fix this issue, the code responsible for timestamp formatting has been modified to display both the date and time, providing a consistent user experience.

## Issue(s)
Closes #5308

## How to test or reproduce
1. In the Rocket.Chat app, quote an old message.
2. Verify that the timestamp in the quoted message now displays both the date and time correctly, as expected.

## Screenshots
N/A

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments
The timestamp issue has been fixed by modifying the timestamp formatting code to display both the date and time. This change ensures consistency with the behavior in the browser version and provides a more accurate representation of the message's timestamp when quoting old messages.

### CODE REVIEW

File: app-supportedversions.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-	"timestamp": "2023-10-11T19:24:10.79990683Z",
+	"timestamp": "2023-10-11T19:24:10.79990683Z",
```
[the remark this answers is not in the document]
Comment (Ansh101112) [reply app-supportedversions.json:3]: changes commited as required

File: app/containers/message/Reply.tsx [hunk 2]
```diff
@@ -96,21 +96,21 @@ interface IMessageReply {
...
-				{attachment.author_name ? (
-					<Text numberOfLines={1} style={[styles.author, { color: themes[theme].auxiliaryTintColor }]}>
-						{attachment.author_name}
-					</Text>
-				) : null}
-				{time ? <Text style={[messageStyles.time, { color: themes[theme].auxiliaryText }]}>{time}</Text> : null}
-				{attachment.title ? <Text style={[styles.title, { color: themes[theme].bodyText }]}>{attachment.title}</Text> : null}
-			</View>
-		);
+	  const time = attachment.message_link && attachment.ts ? moment(attachment.ts).format('YYYY-MM-DD HH:mm:ss') : null; // Modify the format here
```
Comment (diegolmello) [inline app/containers/message/Reply.tsx:99]: I disagree with approach.
It's going to display the full date every time and that takes a lot of space and creates clutter.

You could use a similar approach to RoomItem https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/e004ca39de022e8a2b5ef4f4680c302578f3483c/app/containers/RoomItem/index.tsx#L41, which formats the date relative to new Date()

File: app/containers/message/Reply.tsx [hunk 3]
```diff
@@ -96,21 +96,21 @@ interface IMessageReply {
...
+		<View style={styles.authorContainer}>
+		  {attachment.author_name ? (
+			<Text numberOfLines={1} style={[styles.author, { color: themes[theme].auxiliaryTintColor }]}>
+			  {attachment.author_name}
+			</Text>
+		  ) : null}
+		  {time ? <Text style={[messageStyles.time, { color: themes[theme].auxiliaryText }]}>{time}</Text> : null}
+		  {attachment.title ? <Text style={[styles.title, { color: themes[theme].bodyText }]}>{attachment.title}</Text> : null}
+		</View>
+	  );
```
Comment (diegolmello) [inline app/containers/message/Reply.tsx:110]: You should run `yarn prettier-lint` to get rid of these unnecessary changes.

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/containers/message/Reply.tsx (hunk 2 above)
Comment (Ansh101112) [reply app/containers/message/Reply.tsx]: Please look through it now. I made changes as requested. Thank you for this idea!!
